home/service/search.py

```diff
@@ -76,6 +76,42 @@ def _get_context(self) -> dict[str, Any]:
             "paginator": self.paginator,
             "total_results": self.results.total_results,
             "label_clear_href": label_clear_href,
+            "readable_match_reasons": self._get_match_reason_display_names(),
         }
 
         return context
+
+    def _highlight_results(self):
+        "Take a SearchResponse and add bold markdown where the query appears"
+        string_to_highlight = self.form.cleaned_data.get("query") if self.form.is_valid() else ""
+        highlighted_results = deepcopy(self.results)
+
+        if string_to_highlight == "":
+            return highlighted_results
+
+        else:
+            for result in highlighted_results.page_results:
+                metadata = getattr(result, "metadata")
+                metadata["search_summary"] = metadata.get("search_summary", "").replace(
+                    string_to_highlight, f"**{string_to_highlight}**"
+                )
+                setattr(result, "metadata", metadata)
+
+                name = getattr(result, "description")
+                highlighted_description = name.replace(
```

Comment:
Using string.replace here means the highlighting is going to be case sensitive. We could improve on this by using a case insensitive regex replace:

```
>>> query='bar'
>>> pattern = f'({re.escape(query)})'
>>> re.sub(pattern, r'**\1**', 'foo bar baz BAR', flags=re.IGNORECASE)
'foo **bar** baz **BAR**'
```